ext/liquid_c/vm_assembler.c

```diff
@@ -248,10 +252,9 @@ void vm_assembler_add_write_raw(vm_assembler_t *code, const char *string, size_t
     c_buffer_write(&code->instructions, (char *)string, size);
 }
 
-void vm_assembler_add_write_node(vm_assembler_t *code, VALUE node)
+void vm_assembler_add_write_node(vm_assembler_t *code)
 {
     vm_assembler_write_opcode(code, OP_WRITE_NODE);
-    vm_assembler_write_ruby_constant(code, node);
 }
```

Comment:
The `vm_assembler_t` is recycled after freezing the block and the tag objects live longer than that, so I don't think it can be stored there. The only place where this function is called is in `block_body_add_node` which calls this function (`vm_assembler_add_write_node`) and then writes the tag onto the buffer.